examples/feature_demo/gltf_animations.py

```diff
@@ -0,0 +1,153 @@
+"""
+Animations with glTF
+====================
+
+This example demonstrates how to load a glTF model with animations and play them.
+
+Model from mixamo.com(https://www.mixamo.com)
+
+"""
+
+################################################################################
+# .. note::
+#
+#   To run this example, you need a model from the source repo's example
+#   folder. If you are running this example from a local copy of the code (dev
+#   install) no further actions are needed. Otherwise, you may have to replace
+#   the path below to point to the location of the model.
+
+import os
+from pathlib import Path
+
+try:
+    # modify this line if your model is located elsewhere
+    model_dir = Path(__file__).parents[1] / "data"
+except NameError:
+    # compatibility with sphinx-gallery
+    model_dir = Path(os.getcwd()).parent / "data"
+
+
+################################################################################
+# Once the path is set correctly, you can use the model as follows:
+
+# sphinx_gallery_pygfx_docs = 'animate 4s'
+# sphinx_gallery_pygfx_test = 'run'
+
+import pygfx as gfx
+from wgpu.gui.auto import WgpuCanvas, run
+
+from wgpu.utils.imgui import ImguiRenderer
+from imgui_bundle import imgui
+
+import numpy as np
+
+# Ignore underflow runtime warning for pytest, See: https://github.com/pygfx/pygfx/pull/756#issuecomment-2370715504
+np.seterr(under="ignore")
```

Comment:
i think the error is due to numpy's new strictness in numpy2. 

can we avoid this kind of hackery. i'm happy to give it a look this weekend to see how we can get around the error.

Reply:
We are dealing with this kind of new error in our codebase due to overflows with `uint8`. numpy used to let you just "
```
a = np.zeros(3, dtype='uint8')
a[:] = 0, 128, 256
```

happily, but now will raise warnings and or error.

Reply:
Ok it seems that `np.seterr` is enabled by the test suite:

https://github.com/pygfx/pygfx/blob/main/conftest.py#L37

Reply:
My proposal is to change the line in conftest because there simply are cases where we need to cast float64 to a float32 for the GPU:
```py
np.seterr(all="raise", under="ignore")
```
And then we can remove the `np.seterr()` line in this example.

Makes sense @hmaarrfk?

Reply:
I don’t really follow. But I feel like you understand so that’s good enough for me

Reply:
From what I understand, the code in this PR triggers a floatingpoint underflow in Numpy, because we need to cast to float32 in a few places. I think you're right that this is a general problem that this pr just happens to trigger. Except its not really a problem, but just a fact (GPU's want float32) we need to live with.

So my suggestion is to change the `np.seterr()` call to not trip over these underflows.

Reply:
Yes. That makes sense

Reply:
If there are only a few specific spots where underflows happen, I would prefer to mark those spots with `with np.seterr(...)` locally so that we explicitly allow underflow in those parts of the codebase.

It's very useful to have this check in the test suite because _silently happening_ numerical errors can make it very difficult to trace a problem to its origin.

Reply:
The problem is every transform poses the risk of underflow.

Reply:
We could mark mark the spots where a transform is being stored into a uniform buffer (this is where he cast to float32 happens):
```py
 @callback
    def _update_uniform_buffers(self, transform: AffineBase):
        orig_err_setting = np.seterr(under="ignore")
        self.uniform_buffer.data["world_transform"] = transform.matrix.T
        self.uniform_buffer.data["world_transform_inv"] = transform.inverse_matrix.T
        self.uniform_buffer.update_full()
        np.seterr(**orig_err_setting)
```

There's (at least) one other spot; where the camera transform is set in the std uniform, but for now, the problem is with the transform of the world objects.